### PR TITLE
Reduce JIT icall hash use.

### PR DESCRIPTION
**Pull request by jaykrell**
Reduce JIT icall hash use.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Gradually working toward https://github.com/mono/mono/pull/13947.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Failure doesn't seem related. Hm.

File: mono/metadata/jit-icall-reg.h [hunk 1]
```diff
@@ -350,3 +350,15 @@ mono_check_jit_icall_info (gconstpointer jit_icall_info)
 
 	return (MonoJitICallInfo*)jit_icall_info;
 }
+
+static inline MonoJitICallInfo*
+mono_jit_icall_info_at (int index)
+{
```
Comment (vargaz) [inline mono/metadata/jit-icall-reg.h]: I would call this mono_find_jit_icall_info (). It should also take an enum typed argument.

**Comment by jaykrell** [comment]
std:;vector::at is checked operator [], though that might be obscure, and to me `find` implies more expense to me, like a linear or binary search. `get`? I'm ok either way.

I'm torn as to even use a function vs. direct use of mono_jit_icall_info.array.

The failure seems unrelated, but also specific to this PR. and not flaky.
Like a null pointer calling into loaded LLVM.
So I doubt I will go to master until I resolve that.

And yes I was torn about the `int` vs. `enum`.

`int` is convenient when reaching into a patch info, if I reuse `index. I'll try `enum`, and for patch_info, well, I could add the field, but then there is the read/write, which I guess I'll be pedantic about one way or the other -- like read/write a local int, and copy that to/from the enum. Maybe do a sizeof check to avoid the copy. Minor points though.

We will also have int or uint in the IL stream, in this PR. But ok can cast if needed.
 
In summary:
 - figure out the optional CI failure
 - rename the function to mono_find_jit_icall_info (or get)
 - change int to enum; cast if needed to make that work, hopefully not

File: mono/metadata/jit-icall-reg.h (hunk 1 above)
Comment (jaykrell) [reply mono/metadata/jit-icall-reg.h]: Both done now. We'll see if C++ compiles. :)

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Failure is somehow related.  
I’ll try to repro locally.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
The partial undo, or perhaps the retry, worked.
I will PR each little part of the undo I guess.
And maybe repro locally.

**Comment by jaykrell** [comment]
I will go back before this series of PRs and double check this “llvm” vs. “mono_llvm” stuff.

I recall there is inconsistent naming but only really one set. 

I think we are ok to proceed. I hope CI exercises these paths. But I will look again.

**Comment by jaykrell** [comment]
@monojenkins build failed
